app/bundles/CoreBundle/Shortener/Shortener.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Mautic\CoreBundle\Shortener;
+
+use Mautic\CoreBundle\Helper\CoreParametersHelper;
+
+class Shortener
+{
+    public const SHORTENER_SERVICE = 'shortener_service';
+
+    private CoreParametersHelper $coreParametersHelper;
+
+    /**
+     * @var ShortenerServiceInterface[]
+     */
+    private array $services = [];
+
+    public function __construct(CoreParametersHelper $coreParametersHelper)
+    {
+        $this->coreParametersHelper = $coreParametersHelper;
+    }
+
+    public function addService(string $id, ShortenerServiceInterface $shortener): void
+    {
+        $this->services[$id] = $shortener;
+    }
+
+    public function getService(string $name = null): ShortenerServiceInterface
+    {
+        if (!$name) {
+            $name = $this->coreParametersHelper->get(self::SHORTENER_SERVICE);
+        }
+
+        if (isset($this->services[$name])) {
+            return $this->services[$name];
+        }
+
+        throw new \InvalidArgumentException(sprintf('There is not a shortener service  %s', $name));
+    }
```

Comment:
It looks like this method is never called with the `$name` param.
```suggestion
    public function getService(): ShortenerServiceInterface
    {
        $name = $this->coreParametersHelper->get(self::SHORTENER_SERVICE);

        if (isset($this->services[$name])) {
            return $this->services[$name];
        }

        throw new \InvalidArgumentException(sprintf('There is not a shortener service  %s', $name));
    }
```